vault/plugin_reload_test.go

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+type goodPluginTestBackend struct{}
+
+func (g goodPluginTestBackend) Initialize(ctx context.Context, request *logical.InitializationRequest) error {
+	return nil
+}
+
+func (g goodPluginTestBackend) HandleRequest(ctx context.Context, request *logical.Request) (*logical.Response, error) {
+	return nil, nil
+}
+func (g goodPluginTestBackend) SpecialPaths() *logical.Paths { return nil }
+func (g goodPluginTestBackend) System() logical.SystemView   { return nil }
+func (g goodPluginTestBackend) Logger() hclog.Logger         { return nil }
+func (g goodPluginTestBackend) HandleExistenceCheck(ctx context.Context, request *logical.Request) (bool, bool, error) {
+	return false, false, nil
+}
+func (g goodPluginTestBackend) Cleanup(ctx context.Context)                 {}
+func (g goodPluginTestBackend) InvalidateKey(ctx context.Context, s string) {}
+func (g goodPluginTestBackend) Setup(ctx context.Context, config *logical.BackendConfig) error {
+	return nil
+}
+func (g goodPluginTestBackend) Type() logical.BackendType { return logical.TypeLogical }
+
+type badPluginTestBackend struct{}
```

Comment:
This kind of thing kind of exists in `sdk/plugin/[v5]` but I couldn't get it to work satisfactorily so i implemented specifically what i needed here.